SignalMessaging/categories/UIDevice+featureSupport.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+public extension UIDevice {
+    var supportsCallKit: Bool {
+        return ProcessInfo().isOperatingSystemAtLeast(OperatingSystemVersion(majorVersion: 10, minorVersion: 0, patchVersion: 0))
+    }
+
+    var isIPhoneX: Bool {
+        switch UIScreen.main.nativeBounds.height {
+        case 1136:
+            // iPhone 5 or 5S or 5C
+            return false
+        case 1334:
+            // iPhone 6/6S/7/8
+            return false
+        case 1920, 2208:
+            // iPhone 6+/6S+/7+/8+//
+            return false
+        case 2436:
+            return true
```

Comment:
This will future-proof us better if "iPhone X v2" has same height as iPhone X _and_ other new phones don't.  

`utsname()` would be more conservative. 

Presumably, it'll be easier to update this approach this Summer since the screen sizes of new phones are likely to leak before their machine identifiers do.

If we mis-identify a device, are these the only symptoms:

* the "Return to call" window might have the wrong height.
* nav bars might have the wrong height/layout during a "left" call.

Reply:
This if fine, I'm just trying to think it through.

Reply:
> If we mis-identify a device, are these the only symptoms:
> - the "Return to call" window might have the wrong height.
> - nav bars might have the wrong height/layout during a "left" call.

That's right.